src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/EnumMarshalTests.cs

```diff
@@ -0,0 +1,94 @@
+            RedUpperCase,
+            [EnumExport("RED", EnumValue = ConvertEnum.ToLower)]
+            RedLowerCase,
+            [EnumExport("#ff0000")]
+            RedHex
+        }
+
+        [Fact]
+        public static void MarshalRequestEnums()
+        {
+            Runtime.InvokeJS(@"
+                var dflt = ""Default"";
+                var nostore = ""no-store"";
+                var reload = ""RELOAD"";
+                var nocache = ""nocache"";
+                var force = 3;
+                var onlyif = -3636;
+                App.call_test_method (""SetRequestEnums"", [ dflt, nostore, reload, nocache, force, onlyif ]);
+            ");
+            Assert.Equal(RequestCache.Default, HelperMarshal.requestEnums[0]);
+            Assert.Equal(RequestCache.NoStore, HelperMarshal.requestEnums[1]);
+            Assert.Equal(RequestCache.Reload, HelperMarshal.requestEnums[2]);
+            Assert.Equal(RequestCache.NoCache, HelperMarshal.requestEnums[3]);
+            Assert.Equal(RequestCache.ForceCache, HelperMarshal.requestEnums[4]);
+            Assert.Equal(RequestCache.OnlyIfCached, HelperMarshal.requestEnums[5]);
+        }
+
+        [Fact]
+        public static void MarshalRequestEnumProps()
+        {
+            Runtime.InvokeJS(@"
+                var obj = {};
+                App.call_test_method  (""SetRequestEnumsProperties"", [ obj ]);
+                App.call_test_method  (""SetRequestEnums"", [ obj.dflt, obj.nostore, obj.reload, obj.nocache, obj.force, obj.onlyif ]);
+            ");
+            Assert.Equal(RequestCache.Default, HelperMarshal.requestEnums[0]);
+            Assert.Equal(RequestCache.NoStore, HelperMarshal.requestEnums[1]);
+            Assert.Equal(RequestCache.Reload, HelperMarshal.requestEnums[2]);
+            Assert.Equal(RequestCache.NoCache, HelperMarshal.requestEnums[3]);
+            Assert.Equal(RequestCache.ForceCache, HelperMarshal.requestEnums[4]);
+            Assert.Equal(RequestCache.OnlyIfCached, HelperMarshal.requestEnums[5]);
+        }
+
+        [Fact]
+        public static void MarshalColorEnumProps()
+        {
+            Runtime.InvokeJS(@"
+                var color = {};
+                App.call_test_method  (""SetColorEnumsProperties"", [ color ]);
+                App.call_test_method  (""SetStyleColorEnums"", [ color.Red, color.GreenNumeric, color.RedUpperCase, color.RedLowerCase, color.RedHex, color.RedNoExport ]);
+            ");
+            Assert.Equal(StyleColor.Red, HelperMarshal.styleColorEnums[0]);
+            Assert.Equal(StyleColor.GreenNumeric, HelperMarshal.styleColorEnums[1]);
+            Assert.Equal(StyleColor.RedUpperCase, HelperMarshal.styleColorEnums[2]);
+            Assert.Equal(StyleColor.RedLowerCase, HelperMarshal.styleColorEnums[3]);
+            Assert.Equal(StyleColor.RedHex, HelperMarshal.styleColorEnums[4]);
+            Assert.Equal(StyleColor.RedNoExport, HelperMarshal.styleColorEnums[5]);
+        }
+    }
+}
```

Comment:
Would it be useful to have tests for enums being returned from methods/functions?
How about a test for array of enums being marshaled?

Reply:
Array of enums is not supported right now nor native arrays in general.  They all use a JavaScript object to marshal back and forth.  We can set up a PR for an enhancement to support native arrays but I have not found any examples that actually use an array of enums.  What could be done is create a JavaScript.Array object and fill it out with what is needed to be passed.